tdms/src/main.cpp

```diff
@@ -0,0 +1,67 @@
+    spdlog::set_level(spdlog::level::debug);
+  #elif SPDLOG_ACTIVE_LEVEL == SPDLOG_LEVEL_INFO
+    spdlog::set_level(spdlog::level::info);
+  #endif
+
+  /*
+    There are two cases to consider, when the fdtdgrid matrix is specified in a separate mat file
+    and when it is in the same file as the other matrices.
+  */
+
+  const char *matrixnames[] = {"fdtdgrid","Cmaterial","Dmaterial","C","D","freespace","disp_params","delta","interface","Isource","Jsource","Ksource","grid_labels","omega_an","to_l","hwhm","Dxl","Dxu","Dyl","Dyu","Dzl","Dzu","Nt","dt","tind","sourcemode","runmode","exphasorsvolume","exphasorssurface","intphasorssurface","phasorsurface","phasorinc","dimension","conductive_aux","dispersive_aux","structure","f_ex_vec","exdetintegral","f_vec","Pupil","D_tilde","k_det_obs_global","air_interface","intmatprops","intmethod","tdfield","tdfdir","fieldsample","campssample"};
+  const char *matrixnames_infile[] = {"Cmaterial","Dmaterial","C","D","freespace","disp_params","delta","interface","Isource","Jsource","Ksource","grid_labels","omega_an","to_l","hwhm","Dxl","Dxu","Dyl","Dyu","Dzl","Dzu","Nt","dt","tind","sourcemode","runmode","exphasorsvolume","exphasorssurface","intphasorssurface","phasorsurface","phasorinc","dimension","conductive_aux","dispersive_aux","structure","f_ex_vec","exdetintegral","f_vec","Pupil","D_tilde","k_det_obs_global","air_interface","intmatprops","intmethod","tdfield","tdfdir","fieldsample","campssample"};
+  const char *matrixnames_gridfile[] = {"fdtdgrid"};
+  const char *outputmatrices_all[] = {"Ex_out","Ey_out","Ez_out","Hx_out","Hy_out","Hz_out","x_out","y_out","z_out","Ex_i","Ey_i","Ez_i","Hx_i","Hy_i","Hz_i","x_i","y_i","z_i","vertices","camplitudes","facets","maxresfield","Id","fieldsample","campssample"};
+  const char *outputmatrices[] = {"Ex_out","Ey_out","Ez_out","Hx_out","Hy_out","Hz_out","x_out","y_out","z_out","Ex_i","Ey_i","Ez_i","Hx_i","Hy_i","Hz_i","x_i","y_i","z_i","camplitudes","maxresfield","Id","fieldsample","campssample"};
+  int  matricestosave_all[]   = {0,1,2,3,4,5,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28};
+  int  matricestosave[]   = {0,1,2,3,4,5,10,11,12,13,14,15,16,17,18,19,20,21,23,25,26,27,28};
+  mxArray *plhs[NOUTMATRICES_PASSED];
+  const mxArray *matrixptrs[NMATRICES];
+
+  auto args = ArgumentParser::parse_args(nargs, argv);
+  check_files_can_be_accessed(args);
+
+  //now it is safe to use matlab routines to open the file and order the matrices
+  if(!args.has_grid_filename())
+    openandorder(args.input_filename(), (char **)matrixnames, matrixptrs, NMATRICES);
+  else{
+    openandorder(args.input_filename(), (char **)matrixnames_infile, matrixptrs+1, NMATRICES-1);
+    openandorder(args.grid_filename(), (char **)matrixnames_gridfile, matrixptrs, 1);
+  }
+
+  // decide which derivative method to use (PSTD or FDTD)
+  SolverMethod method = PseudoSpectral; // default
+  if (args.finite_difference())
+    method = SolverMethod::FiniteDifference;
```

Comment:
(Just making myself a note here that this refactor will conflict with the changes from #199, so I don't get confused when I attempt to merge in).

Reply:
:-(

Sorry. I tried to avoid it. But I guess this was always doomed to send you into conflict.